File: 2021/0108_sortedArrayToBST/0818/sortedArrayToBST.rs
impl Solution {
    pub fn sorted_array_to_bst(nums: Vec<i32>) -> Option<Rc<RefCell<TreeNode>>> {
        Self::make_tree(&nums)
    }

    fn make_tree(nums: &[i32]) -> Option<Rc<RefCell<TreeNode>>> {
        if nums.is_empty() { return None }
        let mid = nums.len() / 2;
        Some(Rc::new(RefCell::new(TreeNode {
            val: nums[mid],
            left: Self::make_tree(&nums[..mid]),
            right: Self::make_tree(&nums[mid + 1..]),
        })))
    }
}

File: 2021/0108_sortedArrayToBST/0819/sortedArrayToBST.py
class Solution:
    def sortedArrayToBST(self, nums: List[int]):
        def makeTree(left, right):
            if left > right: return None
            mid = (left + right) // 2
            root = TreeNode(nums[mid])
            root.left = makeTree(left, mid - 1)
            root.right = makeTree(mid + 1, right)
            return root
        return makeTree(0, len(nums) -1)
